fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -823,16 +824,16 @@ public static FHIRSearchContext parseQueryParameters(Class<?> resourceType,
                     SearchParameter referencedParam = getSearchParameter(resourceType, component.getDefinition());
                     compTypes.add(Type.fromValue(referencedParam.getType().getValue()));
                 }
-                queryParameterValues = parseCompositeQueryParameterValuesString(parameterCode, compTypes, queryParameterValueString);
+                queryParameterValues = parseCompositeQueryParameterValuesString(searchParameter, parameterCode, compTypes, queryParameterValueString);
             } else {
-                queryParameterValues = parseQueryParameterValuesString(type, queryParameterValueString);
+                queryParameterValues = parseQueryParameterValuesString(searchParameter, type, queryParameterValueString, null);
             }
         }
         return queryParameterValues;
     }
 
-    private static List<QueryParameterValue> parseCompositeQueryParameterValuesString(String compositeParamCode, List<Type> compTypes,
-            String queryParameterValuesString) throws FHIRSearchException {
+    private static List<QueryParameterValue> parseCompositeQueryParameterValuesString(SearchParameter searchParameter, String compositeParamCode,
+        List<Type> compTypes, String queryParameterValuesString) throws FHIRSearchException {
```

Comment:
the extra indent here is used to distinguish this as a continuation of the method signature and not the first line of the method
```suggestion
            List<Type> compTypes, String queryParameterValuesString) throws FHIRSearchException {
```